curl/tests/client_test.cc

```diff
@@ -469,15 +473,15 @@ namespace leatherman { namespace curl {
              test_impl->test_failure_mode = curl_impl::error_mode::set_url_error;
              REQUIRE_THROWS_AS(test_client.download_file(req, file_path), curl::http_file_download_exception);
```

Comment:
So if we want to isolate just server connection failures, you'd have to catch http_setup_exception first and ignore it, then catch http_request_exception. Or am I catching the http_download_exception?

I don't see an important distinction between request and download. For file exception I can care what file it was trying to operate on. But if we fail to connect to a server, I don't really care where the file was going to be stored.